src/query_plan/mod.rs

```diff
@@ -1 +1,210 @@
+    primary: PrimaryDeferBlock,
+    /// The "deferred" parts of the defer (note that it's a vector). Each of those deferred elements
+    /// will correspond to a different chunk of the response to the client (after the initial
+    /// on-deferred one that is).
+    deferred: Vec<DeferredDeferBlock>,
+}
+
+/// The primary block of a `DeferNode`.
+pub struct PrimaryDeferBlock {
+    /// The part of the original query that "selects" the data to send in that primary response
+    /// once the plan in `node` completes). Note that if the parent `DeferNode` is nested, then it
+    /// must come inside the `DeferredNode` in which it is nested, and in that case this
+    /// sub-selection will start at that parent `DeferredNode.query_path`. Note that this can be
+    /// `None` in the rare case that everything in the original query is deferred (which is not very
+    /// useful  in practice, but not disallowed by the @defer spec at the moment).
+    sub_selection: Option<SelectionSet>,
+    /// The plan to get all the data for the primary block. Same notes as for subselection: usually
+    /// defined, but can be undefined in some corner cases where nothing is to be done in the
+    /// primary block.
+    node: Option<PlanNode>,
+}
+
+/// A deferred block of a `DeferNode`.
+pub struct DeferredDeferBlock {
+    /// References one or more fetch node(s) (by `id`) within `DeferNode.primary.node`. The plan of
+    /// this deferred part should not be started until all such fetches return.
+    depends: Vec<DeferredDependency>,
+    /// The optional defer label.
+    label: Option<NodeStr>,
+    /// Path, in the query, to the `@defer` application this corresponds to. The `sub_selection`
+    /// starts at this `query_path`.
+    query_path: Vec<QueryPathElement>,
+    /// The part of the original query that "selects" the data to send in the deferred response
+    /// (once the plan in `node` completes). Will be set _unless_ `node` is a `DeferNode` itself.
+    sub_selection: Option<SelectionSet>,
+    /// The plan to get all the data for this deferred block. Usually set, but can be `None` for a
+    /// `@defer` application where everything has been fetched in the "primary block" (i.e. when
+    /// this deferred block only exists to expose what should be send to the upstream client in a
+    /// deferred response), but without declaring additional fetches. This happens for @defer
+    /// applications that cannot be handled through the query planner and where the defer cannot be
+    /// passed through to the subgraph).
+    node: Option<PlanNode>,
+}
+
+pub struct DeferredDependency {
+    /// A `FetchNode` ID.
+    id: NodeStr,
+    /// If the `FetchNode` pointed to by `id` has `has_defers` as `true`, and this field is set (to
+    /// the label of one of the defers of said `FetchNode`), then this dependency is strictly for
+    /// the defers with that label.
+    defer_label: Option<NodeStr>,
+}
+
+pub struct ConditionNode {
+    condition: NodeStr,
```

Comment:
For doc-comments: what does this string contain? Is it the name of a boolean variable? An arbitrary expression in some language?

Reply:
It's currently only the name of a boolean variable (although that may change in the future). I'll rename it to `condition_variable` and make its type `Name` to make it clearer, and then later if we need to generalize we can add a generic `condition` with an appropriate type.